packages/react/src/components/Toolbar/ToolbarMenuItem.tsx

```diff
@@ -111,29 +158,134 @@ class ToolbarMenuItem extends UIComponent<WithAsProp<ToolbarMenuItemProps>> {
 
   static defaultProps = {
     as: 'button',
-    accessibility: menuItemBehavior as Accessibility,
+    accessibility: toolbarMenuItemBehavior as Accessibility,
     activeIndicator: 'stardust-checkmark',
+    submenuIndicator: 'stardust-menu-arrow-end',
     wrapper: { as: 'li' },
   }
 
+  static autoControlledProps = ['menuOpen']
+
+  menuRef = React.createRef<HTMLElement>()
+  itemRef = React.createRef<HTMLElement>()
+
   actionHandlers = {
     performClick: event => {
       event.preventDefault()
       this.handleClick(event)
     },
+    openMenu: event => this.openMenu(event),
+    closeAllMenusAndFocusNextParentItem: event => this.closeAllMenus(event),
+    closeMenu: event => this.closeMenu(event),
+    closeMenuAndFocusTrigger: event => this.closeMenu(event),
+    doNotNavigateNextParentItem: event => {
+      event.stopPropagation()
+    },
+    closeAllMenus: event => this.closeAllMenus(event),
   }
 
-  renderComponent({ ElementType, classes, accessibility, unhandledProps, styles }) {
+  openMenu = (e: Event) => {
+    const { menu } = this.props
+    const { menuOpen } = this.state
+    if (menu && !menuOpen) {
```

Comment:
```suggestion
    if (!this.isSubmenuOpen()) {
```

?

Reply:
`!(menu && menuOpen)` is not the same with  `menu && !menuOpen`